streaming/streaming.go

```diff
@@ -0,0 +1,233 @@
+package streaming
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	dataflow "cloud.google.com/go/dataflow/apiv1beta3"
+	datastream "cloud.google.com/go/datastream/apiv1alpha1"
+	datastreampb "google.golang.org/genproto/googleapis/cloud/datastream/v1alpha1"
+	dataflowpb "google.golang.org/genproto/googleapis/dataflow/v1beta3"
+	fieldmaskpb "google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	"github.com/cloudspannerecosystem/harbourbridge/common/utils"
+	"github.com/cloudspannerecosystem/harbourbridge/profiles"
+)
+
+type SrcConnCfg struct {
+	Name     string
+	Location string
+}
+
+type DstConnCfg struct {
+	Name     string
+	Location string
+	Prefix   string
+}
+
+type DatastreamCfg struct {
+	StreamId                    string
+	StreamLocation              string
+	StreamDisplayName           string
+	SourceConnectionConfig      SrcConnCfg
+	DestinationConnectionConfig DstConnCfg
+}
+
+type DataflowCfg struct {
+	JobName  string
+	Location string
+}
+
+type StreamingCfg struct {
+	DatastreamCfg DatastreamCfg
+	DataflowCfg   DataflowCfg
+}
+
+// VerifyAndUpdateCfg checks the fields and errors out if certain fields are empty.
+// It then auto-populates certain empty fields like StreamId and Dataflow JobName.
+func VerifyAndUpdateCfg(streamingCfg *StreamingCfg, dbName string) error {
+	dsCfg := streamingCfg.DatastreamCfg
+	if streamingCfg.DatastreamCfg.StreamLocation == "" {
```

Comment:
This can be dsCfg.StreamLocation since dsCfg is already initialized with streamingCfg.DatastreamCfg